### PR TITLE
Don't modify hostname when running in docker.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/803/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/804/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/806/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/808/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -4167,6 +4167,12 @@ def write_taskqueue_nodes_file()
   # Updates files on this machine with information about our hostname
   # and a mapping of where other machines are located.
   def update_hosts_info()
+    # If we are running in Docker, don't try to set the hostnames.
```
Comment (menivaitsi) [inline AppController/djinn.rb]: hostnames -> hostname

File: scripts/fast-start.sh [hunk 2]
```diff
@@ -105,8 +105,16 @@ elif ${CURL} metadata.google.internal -i |grep 'Metadata-Flavor: Google' ; then
     # As per https://cloud.google.com/compute/docs/metadata.
     PROVIDER="GCE"
 else
-    # Let's assume virtualized cluster.
-    PROVIDER="CLUSTER"
+    # Let's check if this is Docker.
+    if grep docker /proc/1/cgroup > /dev/null ; then
+        # We need to start ssh by hand
```
Comment (menivaitsi) [inline scripts/fast-start.sh]: Add '.' at the end.

File: scripts/fast-start.sh [hunk 3]
```diff
@@ -181,6 +189,9 @@ KEYNAME=$(grep keyname /root/AppScalefile | cut -f 2 -d ":")
 [ -z "${KEYNAME}" ] && { echo "Cannot discover keyname: is AppScale deployed?" ; exit 1 ; }
 
 # Deploy sample app.
+if [ -z "${ADMIN_EMAIL}" ]; then
```
Comment (menivaitsi) [inline scripts/fast-start.sh]: Should we check if this is a docker container instead of if the admin email is set?
Seems more appropriate.

Comment (obino) [inline scripts/fast-start.sh]: Actually this is better: if the email is not set, the next command will fail creating a bad experience. If it's empty it's becasue we set the test flag, so we have a@a.com.

**Pull request merged**